pages/blog/_posts/2020-08-20-building-software-for-a-telecom-core-network.md

```diff
@@ -0,0 +1,157 @@
+
+In an ideal world, VoiceBox would work like this:
+1. The subscriber signs in and enables the desired functionality in VoiceBox
+2. The next time the subscriber receives a voicemail, an event is fired by Working Group Two’s core network, which VoiceBox receives
+3. VoiceBox triggers an SMS/MMS send using our APIs
+4. The subscriber receives an SMS/MMS from the sender “VoiceBox”
+
+In a few months time this ideal world should be reality, but at the time of writing (mid August 2020) we’re missing the events API and the “Send from Product” API. Currently VoiceBox works like this (changes are highlighted):
+1. The subscriber signs in and enables the desired functionality in VoiceBox
+2. The next time the subscriber receives a voicemail, **VoiceBox will discover it by polling**
+3. VoiceBox triggers an MMS send using our APIs
+4. The subscriber receives an **MMS from their own number**
+
+It’s not perfect, but it still demonstrates the potential of the platform. None of this can happen without the subscriber’s consent though, so in the next section we’ll have a look at how that works.
+
+## Obtaining user consent
+
+The app we are building is touching sensitive data, and there is no way we can do that without asking the subscriber if it’s okay. The Working Group Two platform includes an oauth implementation with pin roundtrip authentication, which means that we can be sure that the subscriber (or someone in control of the subscription) has consented to the subscription being controlled by a third party.
+
+When a subscriber opens VoiceBox for the first time, they’re met with a login page, and when they complete the pin challenge they arrive at an oauth consent screen. Here they have to accept the terms of VoiceBox, as well as all the required scopes.
+
+The login is branded to look like the product (notice the pink action button), while the consent screen is branded to look like the operator (in this case, the Swedish operator <a href="https://vimla.se" target="_blank">Vimla</a>, which uses our platform):
+
+<div class="post-images threes">
+    <img src="/img/blog/building-software-for-a-telecom-core-network/msisdn-screen.png" alt="ID login screen">
+    <img src="/img/blog/building-software-for-a-telecom-core-network/pin-screen.png" alt="ID pin screen">
+    <img src="/img/blog/building-software-for-a-telecom-core-network/consent-screen.png" alt="ID consent screen">
+</div>
+
+This follows a standard <a href="https://oauth.net/2/" target="_blank">OAuth</a> flow, which means that when the subscriber taps “Accept”, Working Group Two redirects the subscriber to the third party. This redirect includes a code as a query parameter, which the third party has to use to obtain an access token and a refresh token. This access token gives the third party access to act on behalf of the subscriber. In this case it will let the third party fetch Voicemails and send MMS from the subscriber, so it’s very important to keep it safe.
```

Comment:
```suggestion
This follows a standard <a href="https://oauth.net/2/" target="_blank">OAuth</a> flow, which means that when the subscriber taps “Accept”, Working Group Two redirects the subscriber to the third party, granting them access to act on the users behalf.
```

If this is an "here is our tech" and not a _tutorial_ then I'd rather do without the mini-OAuth 2 explanation.

Reply:
I'll try to shorten it at least.

Reply:
```
When the subscriber taps “Accept”, Working Group Two redirects the subscriber to the third party, which receives an "access token" that allows them to act on behalf of the subscriber. In this case it will let the third party fetch Voicemails and send MMS from the subscriber, so it’s important that the access token is kept safe.
```

Better?